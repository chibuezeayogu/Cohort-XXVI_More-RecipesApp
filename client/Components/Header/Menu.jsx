import React from 'react';
import { withRouter, Link } from 'react-router-dom';

/**
 *
 * @description menu component for unathenticated users
 *
 * @constant
 *
 * @returns {Undefined}
 */
export const Menu = () => (
  <div>
    <nav>
      <div className="nav-wrapper black">
        <a className="brand-logo">
          <img src={'https://res.cloudinary.com/chibuezeayogu/image/upload'+
            '/v1516130489/yy3vdswodkvr3mj5lts1.jpg'}
            className="responsive-img image-radius" 
            alt="" style={{ height: 25, marginRight: 10, marginLeft: 10 }} />
            More Recipes</a>
        <a 
          data-activates="mobile-demo" 
          className="button-collapse">
          <i className="material-icons">menu</i>
        </a>
        <ul className="right hide-on-med-and-down">
          <li>
            <Link to="/">
              <i className="fa fa-home" /> Home
            </Link>
          </li>
          <li>
              <Link to="/signin">
                <i 
                  className="fa fa-sign-in" 
                  aria-hidden="true" />  Login
                </Link>
              </li>
          <li>
            <Link to="/signup">
              <i 
                className="fa fa-user-plus" 
                aria-hidden="true" 
              /> Sign Up
            </Link>
          </li>
        </ul>
        <ul 
          className="side-nav" 
          id="mobile-demo">
          <li>
            <Link to="/">
              <i className="fa fa-home" 
              /> Home
            </Link>
          </li>
          <div className="divider" />
          <li>
            <Link to="/signin">
              <i 
                className="fa fa-sign-in" 
                aria-hidden="true" 
              /> Login
            </Link>
          </li>
          <div className="divider" />
          <li>
            <Link to="/signup">
              <i 
                className="fa fa-user-plus" 
                aria-hidden="true" 
              /> Sign Up
            </Link>
          </li>
          <div className="divider" />
        </ul>
      </div>
    </nav>
  </div>
);
 
export default withRouter(Menu);
