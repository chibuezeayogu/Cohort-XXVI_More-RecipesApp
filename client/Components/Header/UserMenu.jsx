import React, { Component } from 'react';
import PropTypes from 'prop-types';
import jwtDecode from 'jwt-decode';
import { connect } from 'react-redux';
import { SignOut } from '../../action/actionCreators';
import { Link } from 'react-router-dom';
import image from '../../img/logo.jpg';

/**
 * menu component athenticated users
 * @class
 *
 * @extends Component
 *
 */
class UserMenu extends Component {
  /**
   * @description initializes dropdown and button collapse
   *
   * @method
   *
   * @memberOf UserMenu
   *
   * @returns {void}
   *
   */
  componentDidMount() {
    $('.dropdown-button').dropdown({
      inDuration: 300,
      outDuration: 225,
      hover: true,
      belowOrigin: true,
      alignment: 'right',
    });
    $('.button-collapse').sideNav();
  }

  /**
   * @description routes to /if user is logs' out
   *
   * @method
   *
   * @memberOf UserMenu
   *
   * @param {Event} e
   *
   * @returns {void}
   *
   */
  logout(e) {
    e.preventDefault();
    this.props.SignOut();
    localStorage.removeItem('jwtToken');
    this.context.router.history.push('/signin');
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf SignIn
   *
   * @returns {void}
   *
   */
  render() {
    const token = localStorage.getItem('jwtToken');
    const userData = jwtDecode(token);
    return (
      <div>
        <div>
          <nav>
            <div className="nav-wrapper black">
              <a href="#!" className="brand-logo">
                <img
                  src={image}
                  className="responsive-img image-raduis"
                  alt=""
                  style={{ height: 25 }}
                />More Recipes
              </a>
              <a
                href="#!"
                data-activates="mobile-demo"
                className="button-collapse"
              >
                <i className="material-icons">menu</i>
              </a>
              <ul className="right hide-on-med-and-down">
                <li >
                  <Link to="/recipes">
                    <i className="fa fa-home" /> Recipes
                  </Link>
                </li>
                <li>
                  <Link to="/addrecipe">
                    <i className="fa fa-plus" aria-hidden="true" /> Add Recipe
                  </Link>
                </li>
                <li>
                  <Link to="/search">
                    <i className="fa fa-search" aria-hidden="true" /> Search
                  </Link>
                </li>
                <li>
                  <a
                    href="#!"
                    className="dropdown-button"
                    data-activates="dropdown"
                    data-beloworigin="true"
                  > {userData.user.firstName}
                    <i className="material-icons right">arrow_drop_down</i>
                  </a>
                </li>
              </ul>
              <ul
                className="side-nav"
                id="mobile-demo"
              >
                <li>
                  <Link to="/recipes">
                    <i className="fa fa-home" /> Recipes
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/addrecipe">
                    <i className="fa fa-plus" aria-hidden="true" /> Add Recipe
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/search">
                    <i className="fa fa-search" aria-hidden="true" /> Search
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/user/profile">
                    <i className="fa fa-user" aria-hidden="true" /> Profile
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/user/recipes">
                    <i className="fa fa-cutlery" aria-hidden="true" /> My Recipes
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/user/favourites">
                    <i className="fa fa-heart" aria-hidden="true" /> My Favourites
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/" onClick={e => this.logout(e)}>
                    <i className="fa fa-sign-out" aria-hidden="true" /> Logout
                  </Link>
                </li>
              </ul>
            </div>
          </nav>
        </div>

        <ul id="dropdown" className="dropdown-content black-text">
          <div className="divider" />
          <li className="black-text">
            <Link to="/user/profile" className="black-text">
              <i 
                className="fa fa-user black-text" 
                aria-hidden="true" /> My Profile
            </Link>
          </li>
          <div className="divider" />
          <li className="black-text">
            <Link to="/user/recipes" className="black-text">
              <i className="fa fa-cutlery black-text" 
                aria-hidden="true" /> My Recipes
            </Link>
          </li>
          <div className="divider" />
          <li className="black-text">
            <Link to="/user/favourites" className="black-text">
              <i className="fa fa-heart black-text" 
                aria-hidden="true" /> My Favourites
            </Link>
          </li>
          <div className="divider" />
          <li className="black-text">
            <Link to="/" onClick={e => this.logout(e)} className="black-text">
              <i className="fa fa-sign-out black-text"
                aria-hidden="true" /> Logout
            </Link>
          </li>
        </ul>
      </div>
    );
  }
}

UserMenu.contextTypes = {
  router: PropTypes.object.isRequired
};

UserMenu.propTypes = {
  SignOut: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  userData: state.userData,
});

export default connect(mapStateToProps, { SignOut })(UserMenu);
