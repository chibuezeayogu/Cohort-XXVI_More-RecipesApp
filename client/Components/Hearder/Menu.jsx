import React from 'react';
import { Link } from 'react-router-dom';
import image from '../../img/logo.jpg';


/**
 * menu component unathenticated users
 * @constant
 */
const Menu = () => (
  <div>
    <nav>
      <div className="nav-wrapper black">
        <a href="#!" className="brand-logo"><img src={image} className="responsive-img" alt="" style={{ height: 25 }} />More Recipes</a>
        <a href="#!" data-activates="mobile-demo" className="button-collapse">
          <i className="material-icons">menu</i>
        </a>
        <ul className="right hide-on-med-and-down">
          <li><Link to="/"><i className="fa fa-home" /> Home</Link></li>
          <li><Link to="/signin"><i className="fa fa-sign-in" aria-hidden="true" />  Login</Link></li>
          <li><Link to="/signup"><i className="fa fa-user-plus" aria-hidden="true" /> Sign Up</Link></li>
        </ul>
        <ul className="side-nav" id="mobile-demo">
          <li><Link to="/"><i className="fa fa-home" /> Home</Link></li>
          <div className="divider" />
          <li><Link to="/signin"><i className="fa fa-sign-in" aria-hidden="true" /> Login</Link></li>
          <div className="divider" />
          <li><Link to="/signup"><i className="fa fa-user-plus" aria-hidden="true" /> Sign Up</Link></li>
          <div className="divider" />
        </ul>
      </div>
    </nav>
  </div>
);
 
export default Menu;
