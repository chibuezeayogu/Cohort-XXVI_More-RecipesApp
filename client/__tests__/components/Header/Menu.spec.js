import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { Menu } from '../../../Components/Header/Menu';

describe('<Menu />', () => {
  it('render the menu Component', () => {
    const wrapper = shallow(<Menu />);
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Menu />', () => {
  it('shoud call componentDidMount', () => {
    const wrapper = shallow(<Menu />);
    wrapper.instance().componentDidMount();
  });
});

