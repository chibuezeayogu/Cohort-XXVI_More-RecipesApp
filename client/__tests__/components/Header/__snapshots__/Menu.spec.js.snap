// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> render the menu Component 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div>
    <nav>
      <div
        className="nav-wrapper black"
      >
        <a
          className="brand-logo"
        >
          <img
            alt=""
            className="responsive-img image-radius"
            src="https://res.cloudinary.com/chibuezeayogu/image/upload/v1516130489/yy3vdswodkvr3mj5lts1.jpg"
            style={
              Object {
                "height": 25,
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
          />
          More Recipes
        </a>
        <a
          className="button-collapse"
          data-activates="mobile"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
        <ul
          className="right hide-on-med-and-down"
        >
          <li>
            <Link
              id="homeLink"
              replace={false}
              to="/"
            >
              <i
                className="fa fa-home"
              />
               Home
            </Link>
          </li>
          <li>
            <Link
              id="loginLink"
              replace={false}
              to="/signin"
            >
              <i
                aria-hidden="true"
                className="fa fa-sign-in"
              />
                Login
            </Link>
          </li>
          <li>
            <Link
              id="signupLink"
              replace={false}
              to="/signup"
            >
              <i
                aria-hidden="true"
                className="fa fa-user-plus"
              />
               Sign Up
            </Link>
          </li>
        </ul>
        <ul
          className="side-nav"
          id="mobile"
        >
          <li>
            <Link
              id="loginLink"
              replace={false}
              to="/"
            >
              <i
                className="fa fa-home"
              />
               Home
            </Link>
          </li>
          <div
            className="divider"
          />
          <li>
            <Link
              id="loginLink"
              replace={false}
              to="/signin"
            >
              <i
                aria-hidden="true"
                className="fa fa-sign-in"
              />
               Login
            </Link>
          </li>
          <div
            className="divider"
          />
          <li>
            <Link
              id="signupLink"
              replace={false}
              to="/signup"
            >
              <i
                aria-hidden="true"
                className="fa fa-user-plus"
              />
               Sign Up
            </Link>
          </li>
          <div
            className="divider"
          />
        </ul>
      </div>
    </nav>
  </div>,
  "nodes": Array [
    <div>
      <nav>
        <div
          className="nav-wrapper black"
        >
          <a
            className="brand-logo"
          >
            <img
              alt=""
              className="responsive-img image-radius"
              src="https://res.cloudinary.com/chibuezeayogu/image/upload/v1516130489/yy3vdswodkvr3mj5lts1.jpg"
              style={
                Object {
                  "height": 25,
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
            />
            More Recipes
          </a>
          <a
            className="button-collapse"
            data-activates="mobile"
          >
            <i
              className="material-icons"
            >
              menu
            </i>
          </a>
          <ul
            className="right hide-on-med-and-down"
          >
            <li>
              <Link
                id="homeLink"
                replace={false}
                to="/"
              >
                <i
                  className="fa fa-home"
                />
                 Home
              </Link>
            </li>
            <li>
              <Link
                id="loginLink"
                replace={false}
                to="/signin"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-sign-in"
                />
                  Login
              </Link>
            </li>
            <li>
              <Link
                id="signupLink"
                replace={false}
                to="/signup"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-user-plus"
                />
                 Sign Up
              </Link>
            </li>
          </ul>
          <ul
            className="side-nav"
            id="mobile"
          >
            <li>
              <Link
                id="loginLink"
                replace={false}
                to="/"
              >
                <i
                  className="fa fa-home"
                />
                 Home
              </Link>
            </li>
            <div
              className="divider"
            />
            <li>
              <Link
                id="loginLink"
                replace={false}
                to="/signin"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-sign-in"
                />
                 Login
              </Link>
            </li>
            <div
              className="divider"
            />
            <li>
              <Link
                id="signupLink"
                replace={false}
                to="/signup"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-user-plus"
                />
                 Sign Up
              </Link>
            </li>
            <div
              className="divider"
            />
          </ul>
        </div>
      </nav>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Menu />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Menu {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <nav>
            <div
              className="nav-wrapper black"
            >
              <a
                className="brand-logo"
              >
                <img
                  alt=""
                  className="responsive-img image-radius"
                  src="https://res.cloudinary.com/chibuezeayogu/image/upload/v1516130489/yy3vdswodkvr3mj5lts1.jpg"
                  style={
                    Object {
                      "height": 25,
                      "marginLeft": 10,
                      "marginRight": 10,
                    }
                  }
                />
                More Recipes
              </a>
              <a
                className="button-collapse"
                data-activates="mobile"
              >
                <i
                  className="material-icons"
                >
                  menu
                </i>
              </a>
              <ul
                className="right hide-on-med-and-down"
              >
                <li>
                  <Link
                    id="homeLink"
                    replace={false}
                    to="/"
                  >
                    <i
                      className="fa fa-home"
                    />
                     Home
                  </Link>
                </li>
                <li>
                  <Link
                    id="loginLink"
                    replace={false}
                    to="/signin"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-sign-in"
                    />
                      Login
                  </Link>
                </li>
                <li>
                  <Link
                    id="signupLink"
                    replace={false}
                    to="/signup"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user-plus"
                    />
                     Sign Up
                  </Link>
                </li>
              </ul>
              <ul
                className="side-nav"
                id="mobile"
              >
                <li>
                  <Link
                    id="loginLink"
                    replace={false}
                    to="/"
                  >
                    <i
                      className="fa fa-home"
                    />
                     Home
                  </Link>
                </li>
                <div
                  className="divider"
                />
                <li>
                  <Link
                    id="loginLink"
                    replace={false}
                    to="/signin"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-sign-in"
                    />
                     Login
                  </Link>
                </li>
                <div
                  className="divider"
                />
                <li>
                  <Link
                    id="signupLink"
                    replace={false}
                    to="/signup"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user-plus"
                    />
                     Sign Up
                  </Link>
                </li>
                <div
                  className="divider"
                />
              </ul>
            </div>
          </nav>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <nav>
            <div
              className="nav-wrapper black"
            >
              <a
                className="brand-logo"
              >
                <img
                  alt=""
                  className="responsive-img image-radius"
                  src="https://res.cloudinary.com/chibuezeayogu/image/upload/v1516130489/yy3vdswodkvr3mj5lts1.jpg"
                  style={
                    Object {
                      "height": 25,
                      "marginLeft": 10,
                      "marginRight": 10,
                    }
                  }
                />
                More Recipes
              </a>
              <a
                className="button-collapse"
                data-activates="mobile"
              >
                <i
                  className="material-icons"
                >
                  menu
                </i>
              </a>
              <ul
                className="right hide-on-med-and-down"
              >
                <li>
                  <Link
                    id="homeLink"
                    replace={false}
                    to="/"
                  >
                    <i
                      className="fa fa-home"
                    />
                     Home
                  </Link>
                </li>
                <li>
                  <Link
                    id="loginLink"
                    replace={false}
                    to="/signin"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-sign-in"
                    />
                      Login
                  </Link>
                </li>
                <li>
                  <Link
                    id="signupLink"
                    replace={false}
                    to="/signup"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user-plus"
                    />
                     Sign Up
                  </Link>
                </li>
              </ul>
              <ul
                className="side-nav"
                id="mobile"
              >
                <li>
                  <Link
                    id="loginLink"
                    replace={false}
                    to="/"
                  >
                    <i
                      className="fa fa-home"
                    />
                     Home
                  </Link>
                </li>
                <div
                  className="divider"
                />
                <li>
                  <Link
                    id="loginLink"
                    replace={false}
                    to="/signin"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-sign-in"
                    />
                     Login
                  </Link>
                </li>
                <div
                  className="divider"
                />
                <li>
                  <Link
                    id="signupLink"
                    replace={false}
                    to="/signup"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user-plus"
                    />
                     Sign Up
                  </Link>
                </li>
                <div
                  className="divider"
                />
              </ul>
            </div>
          </nav>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Menu />,
}
`;
